spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewritePositionDeleteFiles.java

```diff
@@ -179,7 +179,7 @@ public void testIntPartition() throws Exception {
 
   @Test
   public void testDaysPartitionTransform() throws Exception {
-    createTable("timestamp", "days(partition_col)");
+    createTable("timestamp", "days(`partition_col.type`)");
```

Comment:
Hi, is it possible to leave this test, and just add a new test, in order to keep existing coverage?

For example, we can add partitionCol as an argument to createTable (we can rename existing partitionCol to be partitionTransform)

Reply:
I understand that changing the name of the partition from "partition_col" to "partition_col.type" will not affect the current test coverage. If a test passes with the partition name "partition_col.type", then it must also pass with the partition name "partition_col". 
And the modification you said is to add a test case such as testParitionColWithDot to specifically test the case where the partition column contains ".", without modifying the current test case?

Reply:
Yes , can you please just add a new test?  Its confusing if all the tests now use `` and dot, it's unclear what they are testing for, its better if unit test is more targeted towards one fix/feature.

Reply:
I am aware, we need to change all the methods to take in additional partitionCol parameter, with default value 'partitionCol'

For example:

```
  private void createTable(String partitionType, String partitionCol, String partitionTransform) {
    sql(
        "CREATE TABLE %s (id long, %s %s, c1 string, c2 string) "
            + "USING iceberg "
            + "PARTITIONED BY (%s) "
            + "TBLPROPERTIES('format-version'='2')",
        tableName, partitionCol, partitionType, partitionTransform);
  }

  private void insertData(Function<Integer, ?> partitionValue, int numDataFiles, String partitionCol) throws Exception {
    for (int i = 0; i < numDataFiles; i++) {
      Dataset<Row> df =
          spark
              .range(0, ROWS_PER_DATA_FILE)
              .withColumn(partitionCol, lit(partitionValue.apply(i)))
              .withColumn("c1", expr("CAST(id AS STRING)"))
              .withColumn("c2", expr("CAST(id AS STRING)"));
      appendAsFile(df);
    }
  }

  private List<Object[]> records(String table, String partitionCol) {
    return rowsToJava(
        spark.read().format("iceberg").load(table).sort(partitionCol, "id").collectAsList());
  }


private void testDanglingDelete(String partitionCol) throws Exception {
   ... records(table, partitionCol);
}
```